### PR TITLE
Create base configuration that filters unauthorized access.

### PR DESCRIPTION
**Pull request by florinIacob**

### CODE REVIEW

File: src/main/java/com/myMoneyTracker/app/configuration/WebSecurityConfig.java [hunk 1]
```diff
@@ -6,31 +6,26 @@
...
     }
-
+    
     @Override
-    protected void configure(AuthenticationManagerBuilder authManagerBuilder)
-            throws Exception {
-
+    protected void configure(AuthenticationManagerBuilder authManagerBuilder) throws Exception {
+    
         authManagerBuilder.inMemoryAuthentication().withUser("user").password("password").roles("USER");
```
Comment (tudorgrig) [inline src/main/java/com/myMoneyTracker/app/configuration/WebSecurityConfig.java]: why do we still need this line?

File: src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                response.getOutputStream().print("Unauthorized access!");
+            }
+        }
+    }
+    
+    private boolean isAllowedURL(String url) {
+    
+        boolean isAllowed = false;
+        if (url != null) {
+            if (url.contains("?")) {
```
Comment (tudorgrig) [inline src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java:53]: why do you do the substring if you are using contains?

File: src/main/java/com/myMoneyTracker/app/service/SessionAuthentication.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    private boolean isAuthenticated;
+    private AppUser user;
+    private String sessionToken;
+    private long tokenExpirationTimeMs;
+    
+    public SessionAuthentication(AppUser appUser, String sessionToken) {
+    
+        if (appUser != null) {
+            this.user = appUser;
+            this.name = appUser.getUsername();
```
Comment (tudorgrig) [inline src/main/java/com/myMoneyTracker/app/service/SessionAuthentication.java:32]: why do you store the username if you already store all the user's data?

File: src/main/java/com/myMoneyTracker/controller/AppUserController.java [hunk 4]
```diff
@@ -95,8 +98,9 @@
         return new ResponseEntity<AppUserDTO>(appUserConverter.convertTo(appUser), HttpStatus.OK);
     }
     
-    @RequestMapping(value = "/login", method = RequestMethod.POST)
+    @RequestMapping(value = "/login/", method = RequestMethod.POST)
```
Comment (tudorgrig) [inline src/main/java/com/myMoneyTracker/controller/AppUserController.java]: Why the extra slash? You have nothing coming after that

File: src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import com.myMoneyTracker.app.service.SessionAuthentication;
+
+/**
+ * Class implementing {@link Filter} that will handle unauthorized access.
+ * 
+ * @author Florin
+ */
+public class StatelessAuthenticationFilter implements Filter {
```
Comment (tudorgrig) [inline src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java:23]: Unit tests for the filters need to exist

File: src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java (hunk 2 above)
Comment (florinIacob) [inline src/main/java/com/myMoneyTracker/app/filter/StatelessAuthenticationFilter.java:53]: To clean up a very long path. We need to call contains method many times, and is necessary to 
have a shorter path before performing contains for each allowed string.
Long path example:
'https://www.google.ro/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=whats+happening+if+you+text+a+very+long+path+into+the+URL'

**Pull request merged**